src/core_modules/capture-core/components/Pages/MainPage/TeiWorkingLists/Setup/TeiWorkingListsSetup.component.js

```diff
@@ -1,30 +1,108 @@
+import {
+    dataElementTypes,
+    type TrackerProgram,
+} from '../../../../../metaData';
 import type { Props } from './teiWorkingListsSetup.types';
 import { WorkingLists } from '../../WorkingLists';
 import { useDefaultColumnConfig } from './useDefaultColumnConfig';
 import { useColumns, useDataSource } from '../../WorkingListsCommon';
-import type { TeiWorkingListsColumnConfigs, TeiColumnsMetaForDataFetching } from '../types';
+import type { TeiWorkingListsColumnConfigs, TeiColumnsMetaForDataFetching, TeiFiltersOnlyMetaForDataFetching } from '../types';
 
-const useInjectColumnMetaToLoadList = (defaultColumns, onLoadView) =>
+const useFiltersOnly = ({ enrollment: { enrollmentDateLabel, incidentDateLabel } }: TrackerProgram) => useMemo(() => [{
+    id: 'programStatus',
+    type: dataElementTypes.TEXT,
+    header: i18n.t('Enrollment status'),
+    options: [
+        { text: i18n.t('Active'), value: 'ACTIVE' },
+        { text: i18n.t('Completed'), value: 'COMPLETED' },
+        { text: i18n.t('Cancelled'), value: 'CANCELLED' },
+    ],
+    transformRecordsFilter: (rawFilter: string) => ({
+        programStatus: rawFilter.split(':')[1],
+    }),
+}, {
+    id: 'enrollmentDate',
+    type: dataElementTypes.DATE,
+    header: enrollmentDateLabel,
+    transformRecordsFilter: (filter: Array<string> | string) => {
+        let queryArgs = {};
+        if (Array.isArray(filter)) {
+            queryArgs = filter
+                .reduce((acc, filterPart: string) => {
+                    if (filterPart.startsWith('ge')) {
+                        acc.programStartDate = filterPart.replace('ge:', '');
+                    } else {
+                        acc.programEndDate = filterPart.replace('le:', '');
+                    }
+                    return acc;
+                }, {});
+        } else if (filter.startsWith('ge')) {
+            queryArgs.programStartDate = filter.replace('ge:', '');
+        } else {
+            queryArgs.programEndDate = filter.replace('le:', '');
+        }
+        return queryArgs;
+    },
+}, {
+    id: 'incidentDate',
+    type: dataElementTypes.DATE,
+    header: incidentDateLabel,
+    transformRecordsFilter: (filter: Array<string> | string) => {
+        let queryArgs = {};
+        if (Array.isArray(filter)) {
+            queryArgs = filter
+                .reduce((acc, filterPart: string) => {
+                    if (filterPart.startsWith('ge')) {
+                        acc.programIncidentStartDate = filterPart.replace('ge:', '');
+                    } else {
+                        acc.programIncidentEndDate = filterPart.replace('le:', '');
+                    }
+                    return acc;
+                }, {});
+        } else if (filter.startsWith('ge')) {
+            queryArgs.programIncidentStartDate = filter.replace('ge:', '');
+        } else {
+            queryArgs.programIncidentEndDate = filter.replace('le:', '');
+        }
+        return queryArgs;
+    },
+}, {
```

Comment:
Pretty much the same as the one above, but not putting any effort into generalizing at this point. Probably will do it we need another one with the same structure